### PR TITLE
[systemd] Ensure that podCreateArgs appear last in ExecStartPre=

### PR DESCRIPTION
**Pull request by dcermak**
When creating a new pod without the `--name` flag, e.g.: `podman pod create foobar`
it will get the name `foobar` implicitly and this will be recorded as the in the `podCreateArgs`. Unfortunately, the implicit name only works if it appears as the **last** argument of the startup command.
With 6e2e3a78ed1d05ee5f23f65b814e8135021961dd we started appending the pod security policy to the `startCommand`, resulting in the following `ExecStartPre=` line:
```
/usr/bin/podman pod create --infra-conmon-pidfile %t/pod-foobar.pid --pod-id-file %t/pod-foobar.pod-id foobar --exit-policy=stop
```
This fails to launch, as the `pod create` command expects only a single non-flag parameter, but it assumes that `exit-policy=stop` is a second and terminates immediately instead.

#### Does this PR introduce a user-facing change?

```release-note
- Fix systemd unit generation for pods created without `--name $podName`
```

### CODE REVIEW

**Review by Luap99 — COMMENTED** [review]
Please add unit tests for this, also this fixes #15592
And give it a reasonable release note entry, bug fixes should always have a release note.

@vrothberg PTAL

**Comment by mheon** [comment]
Code LGTM, minus the comment about a test from @Luap99

**Comment by dcermak** [comment]
@mheon @Luap99 I've added a test explicitly for this case, adjusted the existing tests and added a release notes entry.

File: pkg/systemd/generate/pods.go [hunk 1]
```diff
@@ -256,14 +256,14 @@ func generatePodInfo(pod *libpod.Pod, options entities.GenerateSystemdOptions) (
 	return &info, nil
 }
 
-// Unless already specified, the pod's exit policy to "stop".
-func setPodExitPolicy(cmd []string) []string {
+// Determine whether the command array includes an exit-policy setting
+func hasPodExitPolicy(cmd []string) bool {
```
Comment (vrothberg) [inline pkg/systemd/generate/pods.go:260]: Is this rewrite really needed? Could have moved it caller just one line above.

**Review by vrothberg — COMMENTED** [review]
LGTM

File: pkg/systemd/generate/pods.go (hunk 1 above)
Comment (dcermak) [reply pkg/systemd/generate/pods.go:260]: What do you mean by moving the caller a line above?

File: pkg/systemd/generate/pods.go [hunk 2]
```diff
@@ -364,8 +364,10 @@ func executePodTemplate(info *podInfo, options entities.GenerateSystemdOptions)
 			podCreateArgs = append(podCreateArgs, "--replace")
 		}
 
+		if !hasPodExitPolicy(append(startCommand, podCreateArgs...)) {
+			startCommand = append(startCommand, "--exit-policy=stop")
```
Comment (vrothberg) [inline pkg/systemd/generate/pods.go:368]: @dcermak here.  Wouldn't it have been sufficient to just move the old function call here?  My brain is in an afternoon shape, so I may be missing something.

Comment (dcermak) [reply pkg/systemd/generate/pods.go:368]: Yes, that would be possible, but I found the resulting code harder to read as it requires setting a flag whether we have to add the `exit-policy`. I found it simpler to read like this.

Comment (vrothberg) [reply pkg/systemd/generate/pods.go:368]: Thanks! I am OK with the change but usually do not like bug fixes and refactorings in the same commit.

**Comment by dcermak** [comment]
The CI passed now after a temporary fluke with quay.io yesterday. Is there anything missing now? tide complains about a missing `lgtm` label.

**Review by vrothberg — COMMENTED** [review]
/lgtm

**Pull request merged**